src/ragger/navigator/navigator.py

```diff
@@ -258,27 +252,68 @@ def navigate_and_compare(self,
             snaps_tmp_path = self._init_snaps_temp_dir(path, test_case_name, snap_start_idx)
             snaps_golden_path = self._check_snaps_dir_path(path, test_case_name, True)
 
-        if screen_change_before_first_instruction:
+        if enable_auto_wait_screen_change and screen_change_before_first_instruction:
```

Comment:
The `screen_change_before_first_instruction` implicitly relies on Speculos processing the APDU slower than Python entering the `with send_apdu(): navigate` stuff.
Not to be fixed today but to keep in mind